x-pack/solutions/security/plugins/cloud_security_posture/public/components/cloud_security_data_table/cloud_security_data_table.tsx

```diff
@@ -200,8 +200,8 @@ export const CloudSecurityDataTable = ({
    * mode should be passed as a key to the UnifiedDataTable component to force a re-render when the mode changes.
    */
   const computeDataTableRendering = useMemo(() => {
-    // Enable virtualization mode when the table is set to a large page size.
-    const isVirtualizationEnabled = pageSize >= 100;
+    // Enable virtualization mode when the table is set to a large page size and has many rows.
+    const isVirtualizationEnabled = pageSize >= 100 && total >= 100;
```

Comment:
Nice catch! Didn't realize that it was forcing the virtualization even when there were not enough items to be virtualized